dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -126,17 +141,50 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
 
     void AppNotificationManager::Unregister()
     {
+        {
+            auto lock{ m_lock.lock_exclusive() };
+            THROW_HR_IF_MSG(E_FAIL, m_registering, "Register or Unregister currently in progress!");
```

Comment:
This error code should be `ERROR_OPERATION_IN_PROGRESS`.

Reply:
Will update